### PR TITLE
finished the porfolio class and write unittests

### PR DESCRIPTION
**Pull request by JakartaLaw**
Have written unittests, generated docstrings and written the entire module

### CODE REVIEW

File: portfolio/portfolio.py [hunk 1]
```diff
@@ -4,15 +4,81 @@
...
+        self.deposit = deposit
+        self.periods = periods
 
     def calc_annual_return(self):
         """calculates the annual return (plus 1)"""
         return 1 + np.random.normal(loc=self.mu, scale=self.std)
 
-    def
+    def simulate_portfolio(self):
+        """simulates the by for each period inserting the deposit and then calcualte the annual return on the savings"""
```
Comment (JonasPC) [inline portfolio/portfolio.py]: Not sharply formulated

File: portfolio/portfolio.py [hunk 2]
```diff
@@ -4,15 +4,81 @@
...
+        for period in range(0, self.periods + 1):
+
+            # the first period you deposit money and get no return
+            if period is 0:
+                self.simulation_result[period] = self.deposit
+
+            # getting return on investment
+            else:
+                this_year_yield = self.simulation_result[period - 1] * self.calc_annual_return()
+                self.simulation_result[period] = this_year_yield + self.deposit
```
Comment (JonasPC) [inline portfolio/portfolio.py]: I'm not sure i understand why you want to calculate this? I mean, fair enough to calculate the result of the year, but why do you add it with the deposit?

File: portfolio/portfolio.py [hunk 3]
```diff
@@ -4,15 +4,81 @@
...
+        """
+
+        if isinstance(periods, int):
+            result = self.get_single_result(periods)
+        elif isinstance(periods, list):
+            result = self.get_multiple_results(periods)
+        else:
+            raise ValueError("periods must be either int or list")
+
+        return result
```
Comment (JonasPC) [inline portfolio/portfolio.py]: Good

File: tests/test_portfolio.py [hunk 4]
```diff
@@ -5,15 +5,84 @@
...
+
 
-    annual_returns = [p.calc_annual_return() for p in list_1000_portfolio]
+def test_calc_annual_return(list_200_portfolio):
+
+    annual_returns = [p.calc_annual_return() for p in list_200_portfolio]
     mean_annual_return = np.mean(annual_returns)
 
     assert mean_annual_return < 2
     assert mean_annual_return > 0
```
Comment (JonasPC) [inline tests/test_portfolio.py]: Did approx the same. It's not completely accurate, but it works :)

File: portfolio/portfolio.py [hunk 5]
```diff
@@ -4,15 +4,81 @@
...
         self.std = std
-
-    def foo(self):
-        return 'foo'
+        self.deposit = deposit
+        self.periods = periods
 
     def calc_annual_return(self):
         """calculates the annual return (plus 1)"""
         return 1 + np.random.normal(loc=self.mu, scale=self.std)
```
[the remark this answers is not in the document]
Comment (JakartaLaw) [reply portfolio/portfolio.py]: We could make the default values of these parameters 0?

**Pull request merged**